h2o-algos/src/main/java/hex/glrm/GLRM.java

```diff
@@ -241,16 +241,26 @@ private void initLoss() {
     for (int i = 0; i < _ncolA; i++) {
       Vec vi = _train.vec(i);
       GlrmLoss lossi = _lossFunc[i];
-      if (vi.isNumeric()) {
+        // take care of pure numeric columns.  Binary numeric columns has isBinary() = true
+      if (vi.isNumeric() && !vi.isBinary()) {
         if (!lossi.isForNumeric())
           error("_loss_by_col", "Loss function " + lossi + " cannot be applied to numeric column " + i);
+      }  else if (vi.isBinary()) {
+        // Allow numeric loss functions, multi-loss on binary columns too; but not the other way round!
+        if (!lossi.isForBinary() && !lossi.isForNumeric() && !lossi.isForCategorical())
```

Comment:
This should be just `!lossi.isForBinary()`, per description of the method. You may need to tweak the loss functions enum so that `isForBinary()` returns true for all losses that are suitable for binary columns